shared/actions/fs/index.js

```diff
@@ -227,8 +210,11 @@ function* monitorDownloadProgress(key: string, opID: RPCTypes.OpID) {
   }
 }
 
-function* download(action: FsGen.DownloadPayload): Saga.SagaGenerator<any, any> {
-  const {path, intent, key} = action.payload
+function* download(
+  action: FsGen.DownloadPayload | FsGen.ShareNativePayload | FsGen.SaveMediaPayload
+): Saga.SagaGenerator<any, any> {
+  const {path, key} = action.payload
```

Comment:
@songgao why not just generate the key here from the path?

Reply:
`key` is derived from the `path`, and I don't want containers to have to be concerned about it when creating this action. It's really just for internal tracking. Does the latest solution look good to you?

Reply:
I guess you need it for the reducer too. Yep the current solution LGTM.

The reason I wanted the `FsGen` creates to be referenced is that you can then assume without looking at the code what the action is named. If it's just in `Constants`, it _could_ have a different name (even if there's a convention to have it be the same).